app/controllers/catalog_controller.rb

```diff
@@ -319,7 +319,10 @@ def set_form_locals_for_sysprep
   def identify_catalog(id = nil)
     kls = TreeBuilder.get_model_for_prefix(@nodetype) == "MiqTemplate" ? VmOrTemplate : ServiceTemplate
     @record = identify_record(id || params[:id], kls)
-    @tenants_tree = build_tenants_tree if kls == ServiceTemplate # Build the tree with available tenants for the Catalog Item/Bundle
+    if kls == ServiceTemplate
+      add_flash(_("This item is invalid"), :warning) unless @flash_array || @record.template_valid?
```

Comment:
```suggestion
      add_flash(_("This item is invalid"), :warning) unless @flash_array || @record.try(:template_valid?)
```

If you implement it like this, you can pull it out from the branching -> less changed lines (1).